pkg/test/framework/components/echo/echo.go

```diff
@@ -43,14 +44,20 @@ import (
 //     2. Build the source and destination Instances in separate groups and then
 //        call `source.WaitUntilCallable(destination)`.
 type Builder interface {
+	// WithClusters makes subsequent calls to With configure an instance with the same
+	// config to each of the passed clusters. Use methods on the Result returned
+	// by Build to get instances from all clusters.
+	WithClusters([]resource.Cluster) Builder
```

Comment:
I wonder, would it be better to just replace `WithClusters` with:

```golang
WithPerCluster(cfg Config, clusters ...Cluster)
```

This would make it more clear which clusters are targeted for the configuration.

Reply:
See https://github.com/istio/istio/pull/25139